scapy/supersocket.py

```diff
@@ -201,5 +207,123 @@ def recv(self, x=MTU):
         return self.ins.recv(x)
 
 
+class TunTapInterface(SuperSocket):
+    """A socket to act as the host's peer of a tun / tap interface. The
+interface can also be created, using `create=True`.
+
+    """
+    desc = "Act as the host's peer of a tun / tap interface"
+
+    def __init__(self, iface=None, create=False, mode_tun=None, *arg, **karg):
+        self.iface = conf.iface if iface is None else iface
+        self.mode_tun = ("tun" in iface) if mode_tun is None else mode_tun
+        self.created = False
+        if create:
+            self.create()
+        self.closed = True
+        self.open()
+
+    def __enter__(self):
+        return self
+
+    def __del__(self):
+        self.close()
+        self.delete()
+
+    def __exit__(self, *_):
+        pass
+
+    def create(self):
+        """Create the TUN or TAP interface, using system commands (ip on
+Linux, ifconfig on *BSD & Darwin).
+
+        """
+        if self.created:
+            return
+        if LINUX:
```

Comment:
I think that it will be better to create the tun/tap ourselves.

Phil already did some code to do that, see http://www.secdev.org/projects/tuntap_udp/files/tunproxy.py =)

BTW, I wonder if this is needed on Linux.

Reply:
Actually, this code only runs if you specify `create=True`. Do you think it's fine?

Reply:
It is useless on Linux as the interface will be created by `open()`:
```
fd = open("/dev/net/tun", "r+b")
ioctl(fd, 0x400454ca, struct.pack("16sH", "scapy", 0x0001))
os.system("ip link | grep scapy")
```

Reply:
You're right, I've removed the `.create()` & `.delete()` methods.

Reply:
You're right. I've adapted the code in the test to bring the interfaces up before using them and removed the `.create()` and `.delete()` methods. Should be ready now!